test/base-tlsutility.cpp

```diff
@@ -2,11 +2,46 @@
 
 #include "base/tlsutility.hpp"
 #include <BoostTestTargetConfig.h>
+#include <openssl/asn1.h>
+#include <openssl/bn.h>
+#include <openssl/evp.h>
+#include <openssl/obj_mac.h>
+#include <openssl/rsa.h>
+#include <openssl/x509.h>
 #include <utility>
 #include <vector>
 
 using namespace icinga;
 
+static EVP_PKEY* GenKeypair()
+{
+	auto e (BN_new());
+	auto rsa (RSA_new());
+	auto key (EVP_PKEY_new());
+
+	InitializeOpenSSL();
+	BN_set_word(e, RSA_3);
+	BOOST_REQUIRE(RSA_generate_key_ex(rsa, 512, e, nullptr));
+	EVP_PKEY_assign_RSA(key, rsa);
+
+	return key;
+}
+
+static X509* MakeCert(char* issuer, EVP_PKEY* signer, char* subject, EVP_PKEY* pubkey, void(*setTimes)(ASN1_TIME*, ASN1_TIME*))
+{
+	auto cert (X509_new());
+
+	X509_set_version(cert, 2);
+	BN_to_ASN1_INTEGER(BN_new(), X509_get_serialNumber(cert));
+	X509_NAME_add_entry_by_NID(X509_get_issuer_name(cert), NID_commonName, MBSTRING_ASC, (unsigned char*)issuer, -1, -1, 0);
+	setTimes(X509_get_notBefore(cert), X509_get_notAfter(cert));
+	X509_NAME_add_entry_by_NID(X509_get_subject_name(cert), NID_commonName, MBSTRING_ASC, (unsigned char*)subject, -1, -1, 0);
+	X509_set_pubkey(cert, pubkey);
+	BOOST_REQUIRE(X509_sign(cert, signer, EVP_md5()));
```

Comment:
Actually I've tried EVP_md4() before, but newer distros said no. :'(

Reply:
What the... you must be kidding...

That is *exactly* the reason why you shouldn't intentionally use these outdated values here. The same could happen for MD5 and RSA-512. So just the same values we do for regular certificates, like you would do for any sane certificate.

Reply:
1. I'm not lying if you mean that. I can even proof my attempt via GHA history: https://github.com/Icinga/icinga2/actions/runs/6969097918/job/18964321934
2. But yes, admittedly they'd be dropped earlier than what we use in prod.

Reply:
So you have no intention to change this?

Please explain why would anyone want to do this? Who gains anything from using MD5, RSA-512 (and e=3 for that matter)?

If someone sees this code:
* best case scenario: thinks that some code must be from the last century, figures out it was written in 2023 and wonders what the author was thinking.
* worst case scenario: uses this code a template for something where this becomes a serious problem.

Just go ahead and change it to use sane values like the rest of the code. It's really just using different values in a few places, it's not like anything would get more complex because of this.

I really don't understand why you put effort into using worse parameters than the rest of the code and now we are wasting time to discuss this.

Reply:
I've already changed this (before answering btw).